### PR TITLE
feat(react-components): apply default visibility for scene models

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/BND3D-6076

## Description :pencil:

Use the `defaultVisible` property from scene-to-model edges when initializing models through `SceneContainer`. 

## How has this been tested? :mag:

In Fusion

## Test instructions :information_source:

- In 3d-management, select a scene and turn models on/off as desired
- Publish scene
- Use this PR in Unified 3D
- Observe that the models in the same scene has same visibility state as when the scene was stored

## Checklist :ballot_box_with_check:

- [x] I am happy with this implementation.
- [ ] I have performed a self-review of my own code.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have added documentation to new and changed elements; both public and internally shared ones
- [x] I have refactored the code for testability, readability and extendibility to the best of my ability.
- [x] I have listed the JIRA tasks covering remaining work or tech debt related to this PR in the description.

### CODE REVIEW

File: react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts [hunk 1]
```diff
@@ -108,40 +111,36 @@ export class RevealModelsUtils {
...
-            domainObject.setVisibleInteractive(true);
-          }
-          return model;
-        });
-    } else {
-      return await renderTarget.viewer
-        .add360ImageSet('datamodels', {
+    const model = await (async () => {
+      if (options.source === 'events') {
+        return await renderTarget.viewer.add360ImageSet(
```
Comment (pramod-cog) [inline react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts:111]: Why adding model into `DomainObject` is removed?

File: react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+import {
+  isScene360CollectionEdge,
+  isScene3dModelEdge,
+  isSceneConfigurationProperties
+} from './sceneResponseTypeGuards';
+import { type EdgeItem } from '../../data-providers/FdmSDK';
+
+describe('sceneResponseTypeGuards', () => {
+  // Data factory functions for test data generation
+  const createValidSceneConfig = (
```
Comment (pramod-cog) [inline react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts]: Should the helper functions go at the bottom?

File: react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should handle extra properties gracefully', () => {
+      expect(
+        isSceneConfigurationProperties({
+          ...createValidSceneConfig(),
+          extra: 'ignored'
+        } as any)
```
Comment (pramod-cog) [inline react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts]:
```suggestion
        })
```

File: react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+      ).toBe(true);
+
+      expect(
+        isScene3dModelEdge(
+          createValid3dRevisionEdgeItem({
+            ...createValidSceneModel(),
+            extra: 'ignored',
+            another: 42
+          } as any)
```
Comment (pramod-cog) [inline react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts]: Is it possible avoid casting `as any` here?

File: react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+        )
+      ).toBe(true);
+
+      expect(
+        isScene360CollectionEdge(
+          createValid360CollectionEdgeItem({
+            ...createValid360Collection(),
+            extra: 'ignored',
+            another: 42
+          } as any)
```
Comment (pramod-cog) [inline react-components/src/hooks/scenes/sceneResponseTypeGuards.test.ts]: same here, can we avoid casting `as any`?

File: react-components/src/components/RevealToolbar/LayersButton/LayersButton.tsx [hunk 6]
```diff
@@ -15,14 +15,12 @@ export type LayersButtonProps = {
 
 export const LayersButton = ({
   layersState: externalLayersState,
-  setLayersState: setExternalLayersState,
-  defaultLayerConfiguration
```
Comment (pramod-cog) [inline react-components/src/components/RevealToolbar/LayersButton/LayersButton.tsx:19]: remove `defaultLayerConfiguration` from props

File: react-components/src/components/RevealToolbar/LayersButton/LayersButton.viewmodel.tsx [hunk 7]
```diff
@@ -24,7 +24,6 @@ type UseLayersButtonViewModelResult = {
 
 export function useLayersButtonViewModel(
   setExternalLayersState: Dispatch<SetStateAction<LayersUrlStateParam | undefined>> | undefined,
-  defaultLayerConfiguration: DefaultLayersConfiguration | undefined,
```
Comment (pramod-cog) [inline react-components/src/components/RevealToolbar/LayersButton/LayersButton.viewmodel.tsx:27]: Remove `DefaultLayersConfiguration` from import as it is unused

File: react-components/src/components/CadModelContainer/CadModelContainer.tsx [hunk 8]
```diff
@@ -112,10 +114,10 @@ export function CadModelContainer({
         );
       });
       if (viewerModel !== undefined) {
-        return await Promise.resolve(viewerModel as CogniteCadModel);
+        return viewerModel as CogniteCadModel;
       }
       initializingModelsGeometryFilter.current = geometryFilter;
-      return await createCadDomainObject(renderTarget, addModelOptions);
+      return await createCadDomainObject(renderTarget, addModelOptions, defaultVisible);
```
Comment (pramod-cog) [inline react-components/src/components/CadModelContainer/CadModelContainer.tsx:120]: Do think it would be easy to add test as there is a test file already for `defaultVisible` changes?

File: react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts:111]: It has just been moved down a few lines

File: react-components/src/components/CadModelContainer/CadModelContainer.tsx (hunk 8 above)
Comment (haakonflatval-cognite) [reply react-components/src/components/CadModelContainer/CadModelContainer.tsx:120]: Added test :+1:

File: react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts [hunk 9]
```diff
@@ -62,18 +62,17 @@ export class RevealModelsUtils {
...
   public static async addCadModel(
     renderTarget: RevealRenderTarget,
-    options: AddModelOptions
+    options: AddModelOptions,
+    defaultVisible?: boolean
   ): Promise<CogniteCadModel> {
-    const root = renderTarget.root;
-    if (root === undefined) {
-      throw new Error('Root domain object is not set');
-    }
```
Comment (pramod-cog) [inline react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts:70]: Is this necessary, shouldn't we make sure the root domain object is defined?

Comment (haakonflatval-cognite) [reply react-components/src/architecture/concrete/reveal/RevealModelsUtils.ts:70]: It is forbidden from being undefined by the type, so I consider this safe. The RevealRenderTarget also always defines the root on construction, so I don't think this is something we should ever need to worry about
